onnx/cpp2py_export.cc

```diff
@@ -115,13 +115,207 @@ PYBIND11_MODULE(onnx_cpp2py_export, onnx_cpp2py_export) {
+          py::arg("is_homogeneous") = true,
+          py::arg("min_arity") = 1,
+          py::arg("differentiation_category") = OpSchema::DifferentiationCategory::Unknown)
+
+      .def_property_readonly("name", &OpSchema::FormalParameter::GetName)
+      .def_property_readonly("types", &OpSchema::FormalParameter::GetTypes)
+      .def_property_readonly("type_str", &OpSchema::FormalParameter::GetTypeStr)
+      .def_property_readonly("description", &OpSchema::FormalParameter::GetDescription)
+      .def_property_readonly("option", &OpSchema::FormalParameter::GetOption)
+      .def_property_readonly("is_homogeneous", &OpSchema::FormalParameter::GetIsHomogeneous)
+      .def_property_readonly("min_arity", &OpSchema::FormalParameter::GetMinArity)
+      .def_property_readonly("differentiation_category", &OpSchema::FormalParameter::GetDifferentiationCategory)
+      // Legacy camel cased names. We retain them for backward compatibility.
+      // TODO(#5074): Remove these before the 1.16 release.
+      .def_property_readonly(
+          "typeStr",
+          [](const OpSchema::FormalParameter& self) {
+            auto warnings = py::module::import("warnings");
+            warnings.attr("warn")(
+                "OpSchema.FormalParameter.typeStr is deprecated. Use OpSchema.FormalParameter.type_str instead.");
+            return self.GetTypeStr();
+          })
+      .def_property_readonly(
+          "isHomogeneous",
+          [](const OpSchema::FormalParameter& self) {
+            auto warnings = py::module::import("warnings");
+            warnings.attr("warn")(
+                "OpSchema.FormalParameter.isHomogeneous is deprecated. Use OpSchema.FormalParameter.is_homogeneous instead.");
+            return self.GetIsHomogeneous();
+          })
+      .def_property_readonly("differentiationCategory", [](const OpSchema::FormalParameter& self) {
+        auto warnings = py::module::import("warnings");
+        warnings.attr("warn")(
+            "OpSchema.FormalParameter.differentiationCategory is deprecated. Use OpSchema.FormalParameter.differentiation_category instead.");
+        return self.GetDifferentiationCategory();
+      });
+
+  op_schema
+      .def(
+          "__init__",
+          [](OpSchema& self,
+             std::string name,
+             std::string domain,
+             int since_version,
+             std::string doc,
+             std::vector<OpSchema::FormalParameter> inputs,
+             std::vector<OpSchema::FormalParameter> outputs,
+             std::vector<std::tuple<std::string, std::vector<std::string>, std::string>> type_constraints,
+             std::vector<OpSchema::Attribute> attributes) {
+            auto schema = OpSchema().SetName(name).SetDomain(domain).SinceVersion(since_version).SetDoc(doc);
+            // Add inputs and outputs
+            for (auto i = 0; i < inputs.size(); ++i) {
+              const auto& input = inputs[i];
+              schema = schema.Input(
```

Comment:
You are using `OpSchema::operator=` here, while OpSchema::Attribute had a const field and was not assignable.